### PR TITLE
TRITON-2228: Linux CN minimum viable product

### PR DESCRIPTION
**Pull request by bahamat**

### CODE REVIEW

File: lib/lxdclient.js [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+        level: (process.env['LOG_LEVEL'] || 'debug'),
+        name: 'lxd images'
+    });
+    var lxdOpts = {
+        log: log,
+        url: 'https://us.images.linuxcontainers.org'
+        // url: 'https://cloud-images.ubuntu.com/releases'
+    };
+    var wantedImage = 'alpine/3.12';
+    // var wantedImage = 'ubuntu:focal:amd64:default';
```
Comment (danmcd) [inline lib/lxdclient.js:377]: These commented-out ones are for the future, right?

**Review by danmcd — COMMENTED** [review]
You may need someone more JS-savvy to poke through lxdclient.js and lxd.js, so please keep that in mind.

File: package.json [hunk 2]
```diff
@@ -34,8 +34,10 @@
     "posix-getopt": "1.0.0",
     "progbar": "git+https://github.com/trentm/node-progbar.git#a4c56e6",
     "restify": "4.3.1",
+    "restify-clients": "^1.4.0",
+    "restify-errors": "^4.0.0",
     "rimraf": "2.2.6",
-    "sdc-clients": "11.3.0",
+    "sdc-clients": "git+https://github.com/joyent/node-sdc-clients.git#linuxcn",
```
[the remark this answers is not in the document]
Comment (bahamat) [reply package.json:40]: https://github.com/joyent/node-sdc-clients/pull/24

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "forkexec": "1.1.0",
     "glob": "7.0.5",
     "handlebars": "4.0.5",
-    "imgmanifest": "3.1.0",
+    "imgmanifest": "git+https://github.com/joyent/node-imgmanifest.git#linuxcn",
```
[the remark this answers is not in the document]
Comment (bahamat) [reply package.json:21]: https://github.com/joyent/node-imgmanifest/pull/4

File: lib/lxdclient.js (hunk 1 above)
Comment (bahamat) [reply lib/lxdclient.js:377]: ...I actually have no idea what that's about. Presumably that was just Todd doing testing? It seems to work with arbitrary images as-is, but if that's the case then we should be able to just remove that stuff?

**Comment by bahamat** [comment]
@danmcd Like with sdc-designation, we'll need to get node-imgmanifest and node-sdc-clients merged, and then update package.json in this repo. I just missed those when getting everything ready.

File: lib/lxdclient.js (hunk 1 above)
Comment (twhiteman) [reply lib/lxdclient.js:377]: This is for command line (standalone) testing - note that for JS if you see this:
```
if (require.main === module) {
...
}
```
then it's being invoked manually as `node lib/lxdclient.js`.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (bahamat) [reply package.json]: Fixed.

**Comment by danmcd** [comment]
I'd prefer a seasoned JS-er look at lxdclient.js and lxd.js.  If you need me to do the deep-dive, I will, but it'll take longer for me.

File: docs/index.md [hunk 4]
```diff
@@ -961,7 +962,7 @@ CLI tool (from an SDC's IMGAPI):
 
 
 
-## GetImageFile (GET /images/:uuid/file)
+## GetImageFile (GET /images/:uuid/file?filepos=0)
```
Comment (twhiteman) [inline docs/index.md:967]: No such thing as "filepos=0" - this should be removed.
```suggestion
## GetImageFile (GET /images/:uuid/file)
```

File: main.js [hunk 5]
```diff
@@ -86,7 +86,7 @@ function handleArgv() {
         process.exit(0);
     }
     var logSrc = false,
-        logLevel = 'info';
+        logLevel = 'debug';
```
Comment (twhiteman) [inline main.js:89]: Don't think we should be changing log level.

File: main.js [hunk 6]
```diff
@@ -151,7 +151,7 @@ function main() {
 
         function setupWithConfig(_, next) {
             if (!opts.debug && config.logLevel) {
-                log.level(config.logLevel);
+                // log.level(config.logLevel);
```
Comment (twhiteman) [inline main.js:154]:
```suggestion
                log.level(config.logLevel);
```

File: docs/index.md [hunk 7]
```diff
@@ -98,7 +100,7 @@ A summary of fields (details are provided below):
...
 | [origin](#manifest-origin)                                      | UUID    | No                            | No       | The origin image UUID if this is an incremental image.                                                                                                                                          |
-| [files](#manifest-files)                                        | Array   | Yes (if activated)            | No       | An array with a single object describing the image file.                                                                                                                                        |
+| [files](#manifest-files)                                        | Array   | Yes (if activated)            | No       | An array with of object describing the image files.                                                                                                                                             |
```
Comment (danmcd) [inline docs/index.md:103]: "array with of object"... seems like a typo.

**Review by danmcd — COMMENTED** [review]
One last editorial nit.

File: docs/index.md (hunk 7 above)
Comment (bahamat) [reply docs/index.md]: fixed

**Review by danmcd — APPROVED** [review]
Please make sure TRITON-2188 has ALL of the components and how they were tested.

**Pull request merged**